pyrefly/lib/alt/answers.rs

```diff
@@ -79,6 +79,7 @@ pub struct Index {
 struct OverloadedCallee {
     all_overloads: Vec<Callable>,
     closest_overload: Callable,
+    closest_overload_type: Option<Arc<Type>>,
```

Comment:
why'd this an arc? I only see it used in deep_force once. I don't think this helps performance at all?

Reply:
an arc is only more efficient the contents are cloned, but I don't think I see any clones of this (I could be missing them?)